test/lib/godmin/helpers/filters_test.rb

```diff
@@ -13,13 +13,36 @@ def test_filter_form_is_a_bootstrap_form_builder
         end
       end
 
-      # TODO: actually implement this test properly, apparently
-      # assert_select can be used if the helper output is parsed
-      # with HTML::Document first...
-      def test_string_filter_field
-        form = filter_form url: "/" do |f|
+      def test_string_renders_as_string
+        @output_buffer = filter_form url: "/" do |f|
           f.filter_field :foo, as: :string
         end
+
+        assert_select "input[type=text][name=?]", "filter[foo]", 1, "No text input found with name foo"
+      end
+
+      def test_string_filter_field_renders_text_field
+        @output_buffer = filter_form url: "/" do |f|
+          f.string_filter_field :foo, {}, value: "Foobar"
```

Comment:
Should we use `f.filter_field :foo, as: :something` in all tests and consider the `f.string_filter_field` methods etc private? Then we could skip the `test_string_renders_as_string` test and we cover all the `as` options for the `filter_field` method.